Chrysalis-Papilio/Soen-490-Papilio-Android
# Chat 1.0

# General info

**Issue number**: PAPILIO-30

**Task description**: 
 - Implementing chat functionality so that users can join/leave an activity and chat with other people within that activity
 - This is chat 1.0

# Testing
To test this feature you have to:

1. Create an activity using postman using the createChatUser backend branch which ensures that a chat channel gets created.
2. Try to join and leave that newly created activity and see that you have access to the corresponding chat.
3. That's it!

# Notes
- I would test the part where users join/leave an activity where we saw  that there were issues since stream chat users where not yet created, this issue has been fixed on the **createChatUser** branch of the backend. You should be able to join/leave an activity without any issue and the text on the button should show the user state accordingly. If everything works, both the backend and this PR can be merged.
